### DISCUSSION TITLE
GoReleaser in a GitHub workflow to create macOS+arm64

### DISCUSSION THREAD

**Discussion by crgimenes** [hajimehoshi/ebiten#2565] [Q&A]
Hello,
I'm having trouble setting up GoReleaser in a GitHub workflow to create macOS+arm64 releases.

The workflows for other platforms work perfectly, and local compilation works for everything without any problems.

Does anyone know how to configure GitHub workflows to compile programs in Ebiten? Every help is welcome.

This is the repository I'm trying to configure: https://github.com/crgimenes/neko

Accidentally, using Go and Ebiten may end up being the most platform-supported version of Neko in history. :)

**Comment by tinne26** [comment]
If you haven't figured it out already, sharing the actual error that you are getting would be helpful for others to point you in the right direction.

**Reply by crgimenes** [reply]
I didn't have time to dedicate to the problem and still needed to find a good solution.

The alternative I'm considering is to do all the compilation locally and push the releases to GitHub without using Goreleaser or GitHub Actions.

It's not ideal. It would be much better to keep the whole process out of my HQ. But it would be relatively easy to create VMs and make scripts to trigger everything around here.

**Reply by tinne26** [reply]
For github actions in general, see <https://github.com/sinisterstuf/ebitengine-game-template>.

**Reply by crgimenes** [reply]
I believe that the problem is to configure Goreleaser to install the necessary dependencies before trying to compile.

I don't know a way to correctly capture the whole error, so I'm sending a snippet that I think is relevant.

```

release failed after 49s                 error=failed to build for darwin_arm64: exit status 2: go: downloading github.com/hajimehoshi/ebiten/v2 v2.4.16
37
go: downloading github.com/go-gl/glfw/v3.3/glfw v0.0.0-20221017161538-93cebf72946b
38
# github.com/hajimehoshi/ebiten/v2/internal/gamepad
39
Error: ../../../go/pkg/mod/github.com/hajimehoshi/ebiten/v2@v2.4.16/internal/gamepad/gamepad.go:52:10: undefined: newNativeGamepadsImpl
40
# github.com/hajimehoshi/ebiten/v2/internal/glfw

```

Thanks in advance!

**Reply by crgimenes** [reply]
> For github actions in general, see https://github.com/sinisterstuf/ebitengine-game-template.

Very good! I'll read it carefully once I have some time.

**Reply by tinne26** [reply]
I'm not familiarized with goreleaser, but my only hypothesis at the moment would be that it's using go1.16 and it doesn't detect the build tags properly or something like this (EDIT: nevermind, v2.4.16 does use old build tags properly). I can't see anything wrong with Ebitengine or the gamepad source files themselves.

**Reply by crgimenes** [reply]
I'm just not setting up the dependencies and compilation parameters correctly. 
I'm studying how to do this on goreleaser, and I'll let you know as soon as I have an answer.

**Reply by crgimenes** [reply]
Looks like I'm going to need to go back to the idea of using a makefile and compiling locally. I don't really like the idea of using docker. https://goreleaser.com/cookbooks/cgo-and-crosscompiling/
